src/components/NumberInput/NumberInput.vue

```diff
@@ -39,71 +36,53 @@ class Props {
   placeholder = prop<string>({ required: false, default: null })
   required = prop<Boolean>({ default: false })
   modelValue = prop<number | string>({ required: false, default: 0 })
-  value = prop<number | string>({ required: false, default: 0 })
-  width = prop<String>({ default: '' })
+  defaultValue = prop<number | string>({ required: false, default: 0 })
 }
 
-const Component = Options
 @Component({
-  emits: ['update:modelValue'],
-  watch: {
-    value_(val) {
-      if (this.parsedValue > this.parsedMax)
-        return (this.value_ = this.parsedMax)
-      if (this.parsedValue < this.parsedMin)
-        return (this.value_ = this.parsedMin)
-      this.$emit('update:modelValue', val)
-    },
-    modelValue(val) {
-      this.value_ = this.modelValue
-    }
-  }
+  emits: ['update:modelValue']
 })
 export default class NumberInput extends Vue.with(Props) {
-  value_ = this.modelValue || this.value || 0
-  get classList(): string[] {
-    const list = [...(this.disabled ? ['disabled'] : [])]
-    return list
-  }
 
-  get min_(): string {
-    return typeof this.min !== 'string' ? this.min.toString() : this.min
+  private get value(): number {
+    return safeParseFloat(this.modelValue) || safeParseFloat(this.defaultValue) || 0
   }
 
-  get max_(): string {
-    return typeof this.max !== 'string' ? this.max.toString() : this.max
-  }
-  get step_(): string {
-    return typeof this.step == 'number' ? this.step.toString() : this.step
+  private set value(value: number) {
+    if(value < this.parsedMin || value > this.parsedMax) {
+      return
```

Comment:
I believe this does change functionality slightly from what was there originally. If the max is `100` and current value is say `0`, currently if I try to set the value to `500` it would replace that value with `100`. I believe what you have here, it would keep the value at `0`.

Reply:
Good catch. Also added tests for this